lib/functions/general/extensions.sh

```diff
@@ -476,6 +476,14 @@ function enable_extension() {
 			extension_dir="${extension_base_path}" # this is misleading. only directory-based extensions should have this.
 			extension_file="${extension_floating_file}"
 			break
+		else
+			# Search for the extension file in any subdirectory
+			extension_file=$(find "${extension_base_path}" -type f -name "${extension_name}.sh" | head -n 1) # Example format: extensions/network/net-network-manager.sh
+			if [[ -n "${extension_file}" ]]; then
+				# Extract extension dir from file, e.g. from "extensions/network/net-network-manager.sh" the dir "extensions/network/" gets extracted
+				extension_dir="${extension_file%/*}"
+				break
+			fi
```

Comment:
I'm pretty late to this party, but I think this opens up too many possibilities for confusion. eg if someone has `a/a/b.sh` and `a/a/a/b.sh` which is loaded? I appreciate being able to group extensions in sub-directories, but really we should be explicit about them. @ColorfulRhino